terraform/ssh.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package terraform
+
+import (
+	"net"
+	"os"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+// TODO: move this to its separate package
+func sshConn(ip string) (*ssh.Client, error) {
+	conn, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
```

Comment:
Also already mentioned that we could probably check as part of `preFlightCheck` that if `SSH_AUTH_SOCK` is not set we can fail immediately and explain why. This will avoid getting stuck later on during deployment.

Reply:
Done.